src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarmRecorder.java

```diff
@@ -557,6 +579,46 @@ public void perform(@Nonnull hudson.model.Run<?, ?> build, @Nonnull FilePath wor
             writeToLog(log, String.format("Using Project '%s'", projectName));
             Project project = adf.getProject(projectName);
 
+            // Device Selection logic
+            DeviceSelectionConfiguration deviceSelectionConfig = null;
+            ArrayList<DeviceFilter> deviceFilterList = null;
+            DeviceFilter deviceFilter = null;
+
+            if (deviceSelection != null && !deviceSelection.isEmpty()) {
+                deviceSelectionConfig = new DeviceSelectionConfiguration();
+                deviceFilter = new DeviceFilter();
+                deviceFilterList = new ArrayList<DeviceFilter>();
+
+                String deviceSelectionStr = null;
+                deviceSelectionStr = String.format("%s", env.expand(deviceSelection));
+                String[] deviceSelectionList = deviceSelectionStr.split(",");
+                writeToLog(log, String.format("Using Device Selection Filters '%s'", Arrays.toString(deviceSelectionList)));
+
+                for (String deviceFilterString : deviceSelectionList) {
+                    String[] filterElements = deviceFilterString.split(" ", 3);
+                    if (filterElements.length != 3) {
+                        writeToLog(log, String.format("Invalid Device Selection Filter '%s'", deviceFilterString));
+                        build.setResult(Result.FAILURE);
+                        return;
+                    }
+                    deviceFilter  = new DeviceFilter();
+                    deviceFilter.setAttribute(filterElements[0]);
+                    deviceFilter.setOperator(filterElements[1]);
+                    ArrayList<String> testValues = new ArrayList<String>();
+                    testValues.add(filterElements[2]);
```

Comment:
Since value supports as an array - how would you input look like? Also your split.(3) will not work in case customer decides to pass multiple values right?

Reply:
It will. The [2] will take the entire string before the next comma. 
So for example, if the name is "Samsung Galaxy S10" it will take all 3 words. 

For multiple entries, each one becomes a deviceFilter entry.